erpnext/controllers/taxes_and_totals.py

```diff
@@ -272,6 +266,18 @@ def initialize_taxes(self):
 
 			self.doc.round_floats_in(tax)
 
+	def unset_item_wise_tax_details(self):
+		# Keep only item_wise_tax_details that are inserted with `dont_recompute_tax` set to 1.
+		self.doc.item_wise_tax_details = [
+			d for d in self.doc.get("item_wise_tax_details") if d.get("dont_recompute_tax")
+		]
+
+		# If this is a consolidated document, do not unset item_wise_tax_details(it is manually set in consolidated documents).
+		if self.doc.get("is_consolidated"):
+			return
+
+		self.doc._item_wise_tax_details = []
```

Comment:
instead of `is_consolidated` how about filtering the list with `dont_recompute_tax` here